api/app/Http/Validators/UserValidator.php

```diff
@@ -0,0 +1,47 @@
+<?php namespace App\Http\Validators;
+
+use App\Services\Validator;
+
+class UserValidator extends Validator
+{
+    /**
+     * Validation rules.
+     *
+     * @var array
+     */
+    protected $rules = [
+        'email' => 'required|email',
+        'first_name' => 'string',
+        'last_name' => 'string',
+        'phone' => 'string',
+        'mobile' => 'string',
+        'timezone_identifier' => 'timezone',
+        'user_type' => 'string|in:public,admin',
+        '#user_credential.password' => 'min:6'
+    ];
+
+    /**
+     * Model being validated.
+     *
+     * @var string
+     */
+    protected function model()
+    {
+        return 'App\Models\User';
+    }
+
+    /**
+     * Modify the rules for put operations.
+     *
+     * In the case of users, replacement of the entity with a put operation is
+     * not permitted, so the rules is modified according to that.
+     *
+     * @return $this
+     */
+    public function put()
```

Comment:
We should maybe abstract this out to a property of the controller/model to indicate whether total replacement of an entity is allowed.